# Assignment 3

## Submitted by Chaitanya Jindal

#### Q1. Write a Python program to sum all the items in a list.

In [2]:
list = [x for x in range(10)]
sum = 0

for val in list:
    sum += val

print(f"Sum of the values of list {list} = {sum}")

Sum of the values of list [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] = 45


#### Q2: Write a Python program to get the largest number from a list.

In [6]:
import random as r

def largest(numbers):
    largest_num = numbers[0]

    for i in numbers:
        if i > largest_num:
            largest_num = i
        else:
            continue
    
    print(f"The Lagest number from the list {numbers} = {largest_num}")

list = [r.randint(0, 1000) for _ in range(10)]
largest(list)


The Lagest number from the list [316, 340, 103, 226, 84, 381, 210, 357, 525, 838] = 838


#### Q3: Write a Python program to get the smallest number from a list. 

In [8]:
import random as r

def smallest(numbers):
    smallest_num = numbers[0]

    for i in numbers:
        if i < smallest_num:
            smallest_num = i
        else:
            continue
    
    print(f"The Lagest number from the list {numbers} = {smallest_num}")

list = [r.randint(0, 1000) for _ in range(10)]
smallest(list)

The Lagest number from the list [612, 820, 841, 997, 747, 924, 14, 206, 657, 323] = 14


#### Q4: Write a Python program to display the first and last colors from the following list.
               color_list = ["Red","Green","White" ,"Black"] 

In [9]:
color_list = ["Red", "Green", "White", "Black"]

print(f"From the list {color_list}, the first and last colors are [{color_list[0]}, {color_list[-1]}]")

From the list ['Red', 'Green', 'White', 'Black'], the first and last colors are [Red, Black]


#### Q5. Write a program that prints minimum and maximum of five numbers entered by the user.
 

In [19]:
l1 = []
num1 = l1.append(int(input("Enter 1st Number: ")))
num2 = l1.append(int(input("Enter 2nd Number: ")))
num3 = l1.append(int(input("Enter 3rd Number: ")))
num4 = l1.append(int(input("Enter 4th Number: ")))
num5 = l1.append(int(input("Enter 5th Number: ")))

min = max = l1[0]

for i in l1:
    if i > max:
        max = i
    if i< min:
        min = i

print(f"From the list {l1} the min is {min} and max is {max}")

From the list [12, 11, 13, 10, 15] the min is 10 and max is 15


#### Q6: Write a Program to check if a string is palindrome or not 

In [21]:
string1 = input("Enter a string: ").lower()
rev_string = string1[::-1]

if string1 == rev_string:
    print(f"The string {string1} is a palindrome")
else:
    print(f"The string {string1} is not a plaindrome")

The string racecar is a palindrome


#### Q7: Given a nested list. Write a python program to extend it with adding sub list ["h", "i", "j"] in a such a way that it will look like the following list
        Given List: list1 = ["a", "b", ["c", ["d", "e", ["f", "g"], "k"], "l"], "m", "n"]
        Sub List to be added = ["h", "i", "j"]
        Expected output: ['a', 'b', ['c', ['d', 'e', ['f', 'g', 'h', 'i', 'j'], 'k'], 'l'], 'm', 'n']
 

a
b
c, d
   e, f
      g
   k

In [23]:
list1 = ["a", "b", ["c", ["d", "e", ["f", "g"], "k"], "l"], "m", "n"]
sub_list = ["h", "i", "j"]

list1[2][1][2].extend(sub_list)

print(f"Expected list: {list1}")

Expected list: ['a', 'b', ['c', ['d', 'e', ['f', 'g', 'h', 'i', 'j'], 'k'], 'l'], 'm', 'n']


#### Q8: Write a python program for Given a Python list, to find value 20 in the list, and if it is present, replace it with 200. Only update the first occurrence of a value
        list1 = [5, 10, 15, 20, 25, 50, 20]
        Expected output: list1 = [5, 10, 15, 200, 25, 50, 20]
 

In [24]:
list1 = [5, 10, 15, 20, 25, 50, 20]

if 20 in list1:
    index = list1.index(20)
    list1[index] = 200

print(f"The updated list after replacing first number-20 occurence with 200: {list1}")

The updated list after replacing first number-20 occurence with 200: [5, 10, 15, 200, 25, 50, 20]


#### Q9. Write a program to generate the sequence: –5, 10, –15, 20, –25….. upto n, where n is an integer input by the user.

In [31]:
n_th = int(input("Enter the range-n: "))
sequence = [(num*5)*((-1)**num) for num in range(1, n_th+1)]

print(f"The sequence upto {n_th} terms: {sequence}")

The sequence upto 12 terms: [-5, 10, -15, 20, -25, 30, -35, 40, -45, 50, -55, 60]


#### Q10. Write a program to find the sum of 1+ 1/8 + 1/27......1/n3, where n is the number input by the user.

In [33]:
n_th = int(input("Enter the range-n: "))
sum = 0

for i in range(1, n_th+1):
    sum += 1 / (i**3)

print(f"The sum of such pattern upto {n_th} terms is {sum}")

The sum of such pattern upto 12 terms is 1.1988620041787983


#### Q11. Write a program to find the sum of digits of an integer number, input by the user.

In [35]:
num = input("Enter a number: ")
num_list = [int(num) for num in num]

sum = 0

for i in num_list:
    sum += i

print(f"The sum of the digits of the number [{num}] is {sum}")

The sum of the digits of the number [123456789] is 45


#### Q12: Write a python program to display ascii characters from 65 to 90

In [38]:
ascii_list = [num for num in range(65, 91)]

for num in ascii_list:
    print(f"{num} = {chr(num)}")

65 = A
66 = B
67 = C
68 = D
69 = E
70 = F
71 = G
72 = H
73 = I
74 = J
75 = K
76 = L
77 = M
78 = N
79 = O
80 = P
81 = Q
82 = R
83 = S
84 = T
85 = U
86 = V
87 = W
88 = X
89 = Y
90 = Z


#### Q13: Display ascii characters from 48 to 57.
        48 = 0
        49 = 1
        50 = 2
        51 = 3
        52 = 4
        53 = 5
        54 = 6
        55 = 7
        56 = 8
        57 = 9

In [39]:
ascii_list = [num for num in range(48, 58)]

for num in ascii_list:
    print(f"{num} = {chr(num)}")

48 = 0
49 = 1
50 = 2
51 = 3
52 = 4
53 = 5
54 = 6
55 = 7
56 = 8
57 = 9


#### Q14: Display the following output with the help of Ascii character.
        97 = a
        98 = b
        99 = c
        100 = d
        101 = e
        102 = f
        103 = g
        104 = h
        105 = i
        106 = j
        107 = k
        108 = l
        109 = m
        110 = n
        111 = o
        112 = p
        113 = q
        114 = r
        115 = s
        116 = t
        117 = u
        118 = v
        119 = w
        120 = x
        121 = y
        122 = z

In [40]:
ascii_list = [num for num in range(97, 123)]

for num in ascii_list:
    print(f"{num} = {chr(num)}")

97 = a
98 = b
99 = c
100 = d
101 = e
102 = f
103 = g
104 = h
105 = i
106 = j
107 = k
108 = l
109 = m
110 = n
111 = o
112 = p
113 = q
114 = r
115 = s
116 = t
117 = u
118 = v
119 = w
120 = x
121 = y
122 = z


#### Q15: Given a Python list. Write a python program to turn every item of a list into its square
        List1 = [1, 2, 3, 4, 5, 6, 7]
        Expected output: [1, 4, 9, 16, 25, 36, 49]


In [44]:
List1 = [1, 2, 3, 4, 5, 6, 7]

output = [x**2 for x in List1]

print(f"The Expected output for the given list {List1} is {output}")

The Expected output for the given list [1, 2, 3, 4, 5, 6, 7] is [1, 4, 9, 16, 25, 36, 49]


#### Q16: Write a Program to sort alphabetically the words form a string provided by the user. 
    [You can use split() method to split string into a list of words. ]


In [47]:
string = input("Enter a string: ")

words = string.split()

for i in range(len(words)):
    for j in range(0, len(words) - i-1):
        if words[j].lower() > words[j+1].lower():
            words[j], words[j+1] = words[j+1], words[j]

print(f"Entered string: {string} \nSorted words: {words}")

Entered string: Hello, this is a test string, again, Helloo World!!!. 
Sorted words: ['a', 'again,', 'Hello,', 'Helloo', 'is', 'string,', 'test', 'this', 'World!!!.']


#### Q17: write a Program to Remove Punctuations from a String provided by the user. [Hint: use punctuation attribute of string module to get all punctuations 
        (i.e. !"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~ ) ]


In [48]:
import string

str1 = input("Enter a string with punctuations: ")

clean_str = "".join(char for char in str1 if char not in string.punctuation)

print(f"Entered String: {str1} \nOutput Cleaned String: {clean_str}")

Entered String: Hello!! This is Chaitanya Jindal. So nice to meet you, Bye-Bye!! 
Output Cleaned String: Hello This is Chaitanya Jindal So nice to meet you ByeBye


#### Q18. Write a menu driven program to develop SMIS that has options to 
- accept the marks of the student in five major subjects in Class  and display the same.
- calculate the sum of the marks of all subjects. Divide the total marks by number of subjects (i.e. 5), calculate percentage = total marks/5 and display the percentage.
- Find the grade of the student as per the following criteria:

| Criteria                                 | Grade    |
|------------------------------------------|----------|
| percentage > 85                          | A        |
| percentage < 85 && percentage >= 75      | B        |
| percentage < 75 && percentage >= 50      | C        |
| percentage < 50 && percentage >= 30      | D        |
| percentage < 30                          | Reappear |


In [52]:
marks = []
total = 0

while True:
    print("----- SMIS Menu -----")
    print("1. Accept and Display Marks")
    print("2. Calculate Total and Percentage")
    print("3. Calculate Grade")
    print("4. Exit")

    choice = int(input("Enter your choice (1-4): "))

    if choice == 1:
        marks = []
        print("Enter marks for 5 major subjects: ")
        for i in range(5):
            m = float(input(f"Subject {i+1}: "))
            marks.append(m)
            total += m
        print(f"Marks entered: {marks}")

    elif choice == 2:
        if not marks:
            print("Please enter marks first (Option 1).")
        else:
            percentage = total/ 5
            print(f"Total Marks: {total}")
            print(f"Percentage: {percentage:.2f}%")
    
    elif choice == 3:
        if not marks:
            print("Please enter marks first (Option 1).")
        else:
            if percentage >= 85:
                grade = "A"
            elif percentage >= 75:
                grade = "B"
            elif percentage >= 50:
                grade = "C"
            elif percentage >= 30:
                grade = "D"
            else:
                grade = "Reappear"
            print(f"Percentage: {percentage:.2f}% - Grade: {grade}")

    elif choice == 4:
        print("Exiting SMIS...")
        break

    else:
        print("Invalid Choice!! Please try again!")

----- SMIS Menu -----
1. Accept and Display Marks
2. Calculate Total and Percentage
3. Calculate Grade
4. Exit
Enter marks for 5 major subjects: 
Marks entered: [78.0, 87.0, 89.0, 98.0, 65.0]
----- SMIS Menu -----
1. Accept and Display Marks
2. Calculate Total and Percentage
3. Calculate Grade
4. Exit
Total Marks: 417.0
Percentage: 83.40%
----- SMIS Menu -----
1. Accept and Display Marks
2. Calculate Total and Percentage
3. Calculate Grade
4. Exit
Percentage: 83.40% - Grade: B
----- SMIS Menu -----
1. Accept and Display Marks
2. Calculate Total and Percentage
3. Calculate Grade
4. Exit
Invalid Choice!! Please try again!
----- SMIS Menu -----
1. Accept and Display Marks
2. Calculate Total and Percentage
3. Calculate Grade
4. Exit
Exiting SMIS...


#### Q19: Write a python program to print the Following:
        1
        2   1
        3  2  1


In [58]:
for i in range(1, 4):
    for j in range(i, 0, -1):
        print(j, end="  ")
    print("\n")

1  

2  1  

3  2  1  



#### Q20: WAP to print the following asterisk pattern:
        *
        **
        ***
        ****
        *****
        ******

In [62]:
n_th = int(input("Enter the number of rows for the pattern: "))

for i in range(1, n_th+1):
    print("*"*i)

*
**
***
****
*****
******


#### Q21: WAP to create a function triangle to print the following asterisk triangle pattern:
        *
        **
        ***
        ****
        ***
        **
        *

In [79]:
n_th = float(input("Enter the number of rows for the pattern: "))

for i in range(1, int(n_th)):
    print("*"*i)

for i in range(int(n_th), 0, -1):
    print("*"*i)

*
**
***
****
*****
****
***
**
*


#### Q22: Write a python program to print following multiplication table on the screen
        |   | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 |
        |---|---|---|---|---|---|---|---|---|---|----|
        |1  |1  |2  |3  |4  |5  |6  |7  |8  |9  |10 |
        |2  |2  |4  |6  |8  |10 |12 |14 |16 |18 |20 |
        |3  |3  |6  |9  |12 |15 |18 |21 |24 |27 |30 |
        |4  |4  |8  |12 |16 |20 |24 |28 |32 |36 |40 |
        |5  |5  |10 |15 |20 |25 |30 |35 |40 |45 |50 |
        |6  |6  |12 |18 |24 |30 |36 |42 |48 |54 |60 |
        |7  |7  |14 |21 |28 |35 |42 |49 |56 |63 |70 |
        |8  |8  |16 |24 |32 |40 |48 |56 |64 |72 |80 |
        |9  |9  |18 |27 |36 |45 |54 |63 |72 |81 |90 |
        |10 |10 |20 |30 |40 |50 |60 |70 |80 |90 |100|

In [18]:
print("|     |", end="")
for i in range(1, 11):
    print(f"  {i:<2} |", end="")
print("\n|" + " --- |" * 11)

for i in range(1, 11):
    print(f"| {i:<3} |", end="")
    
    for j in range(1, 11):
        product = i * j
        print(f" {product:<3} |", end="")
    
    print()

|     |  1  |  2  |  3  |  4  |  5  |  6  |  7  |  8  |  9  |  10 |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 1   | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10  |
| 2   | 2   | 4   | 6   | 8   | 10  | 12  | 14  | 16  | 18  | 20  |
| 3   | 3   | 6   | 9   | 12  | 15  | 18  | 21  | 24  | 27  | 30  |
| 4   | 4   | 8   | 12  | 16  | 20  | 24  | 28  | 32  | 36  | 40  |
| 5   | 5   | 10  | 15  | 20  | 25  | 30  | 35  | 40  | 45  | 50  |
| 6   | 6   | 12  | 18  | 24  | 30  | 36  | 42  | 48  | 54  | 60  |
| 7   | 7   | 14  | 21  | 28  | 35  | 42  | 49  | 56  | 63  | 70  |
| 8   | 8   | 16  | 24  | 32  | 40  | 48  | 56  | 64  | 72  | 80  |
| 9   | 9   | 18  | 27  | 36  | 45  | 54  | 63  | 72  | 81  | 90  |
| 10  | 10  | 20  | 30  | 40  | 50  | 60  | 70  | 80  | 90  | 100 |
